apps/laboratory/src/utils/SiweUtils.ts

```diff
@@ -1,20 +1,19 @@
-import { SiweMessage } from 'siwe'
 import { getCsrfToken, signIn, signOut, getSession } from 'next-auth/react'
 import type { SIWEVerifyMessageArgs, SIWECreateMessageArgs, SIWESession } from '@web3modal/siwe'
 import { createSIWEConfig } from '@web3modal/siwe'
+import { formatMessage } from '@walletconnect/utils'
+import { WagmiConstantsUtil } from '../utils/WagmiConstants'
 
 export const siweConfig = createSIWEConfig({
-  createMessage: ({ nonce, address, chainId }: SIWECreateMessageArgs) =>
-    new SiweMessage({
-      version: '1',
-      domain: window.location.host,
-      uri: window.location.origin,
-      address,
-      chainId,
-      nonce,
-      // Human-readable ASCII assertion that the user will sign, and it must not contain `\n`.
-      statement: 'Sign in With Ethereum.'
-    }).prepareMessage(),
+  // We don't require any async action to populate params but other apps might
```

Comment:
Can we explain it a bit? In which situations the apps might require await the `getMessageParams` function? My concern is that users shouldn't confuse about how to call it and make sure about the type definitions since when we make it async, the return type will be also `Promise<..>`

Reply:
Any case where you might want to build dynamic `statement` or `uri` or `resources` with recaps or even provide `nonce` with the params instead of using the `getNonce` client method like
```
getMessageParams: async () => ({
    domain: window.location.host,
    uri: `${window.location.origin}/login/${await getSomeDynamicToken()}`,
    chains: WagmiConstantsUtil.chains.map(chain => chain.id),
    statement: 'Please sign with your account',
    nonce: await getCsrfToken()
  }),
```

Reply:
I think this could be of type `()=>(Promise<MessageParams> | MessageParams)`

So both sync and async fn could be allowed.